### DISCUSSION TITLE
Error: P1000

### DISCUSSION THREAD

**Discussion by mynameundefined** [prisma/prisma#12120] [Q&A] [answered]

**Comment by austincrim** [comment, marked as answer]
Hey @Ivanhai 👋 ,

Are you using your full connection string in your `schema.prisma` or are you using an environment variable? 

Do you have any special characters in your username/password?

**Reply by mynameundefined** [reply]
i am using  an environment variable and dont have special characters

**Reply by austincrim** [reply]
Are you certain your environment variable is set correctly and the corresponding name is in your `schema.prisma`? e.g.

```.env
# .env
DATABASE_URL="my connection string"
```
```prisma
// schema.prisma
datasource db {
  provider = "postgres"
  url      = env("DATABASE_URL")
}
```

If it is still not working, I would recommend opening an [issue](https://github.com/prisma/prisma/issues/new?assignees=&labels=kind%2Fbug&template=bug_report.yml) with all of your relevant environment information.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
